### DISCUSSION TITLE
OpenAPI v3 import - debugging "factory is not a function" error message

### DISCUSSION THREAD

**Discussion by IvanRibakov** [Kong/insomnia#3722] [Help] [answered]
- Insomnia Version: 6.4.2
- Operating System: macOS 10.14.4

## Details

Tried to use new OpenAPI v3 importer, but as I tried to load it from file I got a brief `factory is not a function` error message:

![image](https://user-images.githubusercontent.com/3639958/57281280-f3e2f780-70aa-11e9-93cd-4303e349100b.png)

I'm guessing this is probably due to the current lack of full OpenAPI v3 support, but I'm not sure how to go about determining which particular part of the OpenAPI v3 document breaks the import. Any suggestions where to start looking for the error cause?

**Comment by welcome** [comment]
👋 Thanks for opening your first issue! If you're reporting a 🐞 bug, please make sure
you include steps to reproduce it. If you're requesting a feature 🎁, please provide real
use cases that would benefit. 👪

To help make this a smooth process, please be sure you have first read the
[contributing guidelines](https://github.com/getinsomnia/insomnia/blob/master/CONTRIBUTING.md).

**Comment by gschier** [comment, marked as answer]
It looks like the error would be coming from here: 

https://github.com/getinsomnia/insomnia/blob/7a549310470fc17851a5358e740dcfae85249d2c/packages/insomnia-importers/src/importers/openapi3.js#L333

It looks like maybe your import has a type that is not in this map?

https://github.com/getinsomnia/insomnia/blob/7a549310470fc17851a5358e740dcfae85249d2c/packages/insomnia-importers/src/importers/openapi3.js#L287-L315

I will add a better error message there.
